### PR TITLE
fix several bugs

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
[AB#1892692](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1892692): error message is not clear enough
[AB#1891898](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1891898): [Test] Pops up com.azure.core.management.exception.ManagementException while deploying spring cloud with a creating service
[AB#1896364](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1896364): [Test]UI freezes when deploy spring cloud
[AB#1896376](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1896376): npe on clicking OK if there are no valid spring cloud service
[AB#1896359](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1896359): [Test] Pops up NPE when searching with GET and showing hash typed key value with Redis
[AB#1894293](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1894293): [Test] When click Open by Database Tools/Show Properties , the status bar shows '<unknown>'
[AB#1895989](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1895989): Draft Spring Cloud App in IntelliJ Spring Cloud run configuration could not be loaded when reopen run configuration

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfigurationState.java [hunk 1]
```diff
@@ -66,10 +67,17 @@ public SpringCloudDeploymentConfigurationState(Project project, SpringCloudDeplo
...
         consoleView.attachToProcess(processHandler);
-        final Disposable subscribe = Mono.fromCallable(() -> this.execute(messager))
-                .doOnTerminate(processHandler::notifyComplete)
-                .subscribeOn(Schedulers.boundedElastic())
-                .subscribe((res) -> messager.success("Deploy succeed!"), messager::error);
+        final Disposable subscribe = Mono.fromCallable(() -> {
+                try {
+                    return this.execute(messager);
+                } catch (final Exception e) {
+                    messager.error(e);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfigurationState.java]: Will we still show deploy succeed if we catch exception here?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java [hunk 2]
```diff
@@ -24,11 +24,19 @@
...
+        super();
+        this.setDefaultValue(0);
+    }
+
     @Nullable
     @Override
     public Integer getValue() {
         final String text = getText();
-        if (StringUtils.isBlank(text) || !StringUtils.isNumeric(text)) {
+        if (StringUtils.isBlank(text)) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java]: do we also need to check isRequired here?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java [hunk 3]
```diff
@@ -24,11 +24,19 @@
     @Setter
     private Integer maxValue;
 
+    public AzureIntegerInput() {
+        super();
+        this.setDefaultValue(0);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java]: We may also need to check whether default value fit min/max limitation

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java]: no need. validation will do the check.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java (hunk 3 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureIntegerInput.java]: user can/should set default value mannually if min/max are set.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfigurationState.java [hunk 4]
```diff
@@ -68,18 +66,18 @@ public SpringCloudDeploymentConfigurationState(Project project, SpringCloudDeplo
...
-                return this.execute(messager);
+                this.execute(messager);
+                messager.success("Deploy succeed!");
             } catch (final Exception e) {
                 messager.error(e);
             }
-            return Optional.empty();
         };
-        final Disposable subscribe = Mono.fromCallable(execute)
+        final Disposable subscribe = Mono.fromRunnable(execute)
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfigurationState.java:77]: May change to `runOnPoolThread` with `try catch finally`

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**